lambda_src/lambda_function.py

```diff
@@ -155,30 +220,80 @@ def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
             'statusCode': 200,
             'body': b64encode(compress(data_dumps.encode())).decode(),
             'isBase64Encoded': True,
-            'headers': {'Content-Encoding': 'gzip'}
+            'headers': {'Content-Encoding': 'gzip'},
         }
+        end_time = timer()
+        if response and (end_time - start_time) > 20:
+            LOG.debug('Storing the response in lock cache.')
+            try:
+                finish_batch_processing(batch_id, response)  # write the response
+            except ClientError as ce:
+                if ce.response['Error']['Code'] == 'ValidationException':
+                    LOG.error(ce)
+                    error_dumps = dumps(
+                        {
+                            'data': [
+                                [
+                                    rn,
+                                    {
+                                        'error': f'Response size ({len(dumps(response))} bytes) too large to be stored in the backend.',
```

Comment:
"response size for batch id ... is too large (... bytes gzip'd). the batch size is {len(...)} and this row's response size is ... bytes"